### PR TITLE
node fixes: correct a call and improve variable names

### PR DESCRIPTION
**Pull request by telephon**
- The method “subclassMustImplementThisMethod” has never existed.
- Renamed `addNum` to `addActionID` and replaced `inTarget` by a direct
conversion.
- brought code to a more standard form.

### CODE REVIEW

File: SCClassLibrary/Common/Control/Node.sc [hunk 1]
```diff
@@ -246,23 +248,23 @@ AbstractGroup : Node {
...
 
 	newMsg { arg target, addAction = \addToHead;
-		var addNum, inTarget;
+		var addActionID;
 		// if target is nil set to default group of server specified when basicNew was called
-		inTarget = (target ? server.defaultGroup).asTarget;
-		addNum = addActions[addAction];
-		(addNum < 2).if({ group = inTarget; }, { group = inTarget.group; });
-		^[this.class.creationCmd, nodeID, addNum, inTarget.nodeID]
+		target = target.asTarget;
```
Comment (brianlheim) [inline SCClassLibrary/Common/Control/Node.sc]: sorry if i'm misunderstanding, but why did you remove the nil check here?

Comment (brianlheim) [reply SCClassLibrary/Common/Control/Node.sc]: never mind, I understand now!

File: SCClassLibrary/Common/Control/Node.sc [hunk 2]
```diff
@@ -463,12 +469,10 @@ Synth : Node {
 	}
 
 	newMsg { arg target, args, addAction = \addToHead;
-		var addNum, inTarget;
-		addNum = addActions[addAction];
-		// if target is nil set to default group of server specified when basicNew was called
```
Comment (brianlheim) [inline SCClassLibrary/Common/Control/Node.sc]: maybe better to leave this comment - it's not immediately clear that this is what happens when you call asTarget on nil (as I just showed :P)

File: SCClassLibrary/Common/Control/Node.sc [hunk 3]
```diff
@@ -145,18 +147,18 @@ Node {
 			# cmd, argnodeID, parent, prev, next, isGroup, head, tail = msg;
 			// assuming its me ... if(nodeID == argnodeID)
 			Post << if(isGroup == 1, "Group:" , "Synth:") << nodeID << Char.nl
-				<< "parent  : " << parent << Char.nl
-				<< "prev : " << prev << Char.nl
-				<< "next :" << next << Char.nl;
+			<< "parent  : " << parent << Char.nl
```
Comment (brianlheim) [inline SCClassLibrary/Common/Control/Node.sc]: IMO this makes the code less readable, regardless of it being the IDE's indentation behavior

**Comment by brianlheim** [comment]
overall lgtm. i think this is safe to merge for 3.9.0 since no functionality is altered.

File: SCClassLibrary/Common/Control/Node.sc (hunk 2 above)
Comment (telephon) [reply SCClassLibrary/Common/Control/Node.sc]: In all other cases, there are no comments. This behavior is just the standard for `asTarget` (it's documented in the "asTarget" helpfile).

File: SCClassLibrary/Common/Control/Node.sc (hunk 3 above)
Comment (telephon) [reply SCClassLibrary/Common/Control/Node.sc]: ah yes, I know, it's a little annoying. I wish autoindent were a little more tolerant sometimes.  I was just too lazy to undo it, but will do it now.

**Comment by telephon** [comment]
yes still good to have a careful look, because it's easy to overlook something here.

File: SCClassLibrary/Common/Control/Node.sc (hunk 2 above)
Comment (brianlheim) [reply SCClassLibrary/Common/Control/Node.sc]: ok! line 263 above still has it. maybe it should be removed there too for consistency then

**Comment by telephon** [comment]
I could change the `arg x` syntax to `|x|` syntax, if that is a good moment for it.

**Comment by brianlheim** [comment]
> I could change the arg x syntax to |x| syntax, if that is a good moment for it.

probably an action item best left for later, this PR is a good size as it is.

File: SCClassLibrary/Common/Control/Node.sc [hunk 4]
```diff
@@ -146,14 +146,16 @@ Node {
...
-			});
+			Post
+				<< if(isGroup == 1, "Group:\t" , "Synth:\t") << nodeID << Char.nl
+				<< "parent:\t" << parent << Char.nl
+				<< "prev:\t" << prev << Char.nl
+				<< "next:\t" << next << Char.nl;
+			if(isGroup == 1) {
+				Post
+				<< "head:\t" << head << Char.nl
+				<< "tail:\t" << tail << Char.nl << Char.nl;
```
Comment (brianlheim) [inline SCClassLibrary/Common/Control/Node.sc]: these 2 lines should also be indented by 1 more tab

Comment (telephon) [reply SCClassLibrary/Common/Control/Node.sc]: I thought it was nice that you can just read them downwards.

Comment (brianlheim) [reply SCClassLibrary/Common/Control/Node.sc]: but then the indentation style is inconsistent, i think it is less clear this way. when code is more predictably formatted, it takes less effort to read. the rule you're establishing here does not have much precedent and is also IMO difficult to explain if you wanted it to be followed elsewhere.
